### ISSUE TITLE
Bug: `settings.diagnostics.ignoreCode` is ignored

### ISSUE THREAD

**Issue by chrisgrieser** [pmizio/typescript-tools.nvim#233]
When I have this in my config:

```lua
require("typescript-tools").setup {
    settings = {
        diagnostics = { 
             ignoredCodes = { 2451 } 
        },
    }
}
```

That diagnostic is still reported when using `typescript-tools.nvim`. Using the very same config for tsserver via nvim-lsp-config, `settings.diagnostics.ignoreCode` works fine.

**Comment by pmizio** [comment]
@chrisgrieser Everything you need to resolve this is in [README](https://github.com/pmizio/typescript-tools.nvim/blob/c43d9580c3ff5999a1eabca849f807ab33787ea7/README.md?plain=1#L169).
Long story short we use native neovim shenanigans to implement this feature. In your case it'll be:
```lua
local api = require("typescript-tools.api")
require("typescript-tools").setup {
  handlers = {
    ["textDocument/publishDiagnostics"] = api.filter_diagnostics { 2451 },
  },
}
```

**Comment by chrisgrieser** [comment]
well, at the very least, then this sentence from the readme is wrong:
> The parameters passed into the setup function are also passed to the standard nvim-lspconfig server setup, **allowing you to use the same settings here**.

"The very same settings" cannot be used here. I didn't go further down the readme, since that part of the README led me to believe that `settings.diagnostics.ignoreCode` should be working
